<html class="gr__amp_pharm_mssm_edu">

<head>
    <meta http-equiv="Content-Type" content="text/html; charset=windows-1252">
    <style type="text/css" abt="234"></style>
</head>

<body data-gr-c-s-loaded="true">
    <div id="the-basics-content">
        <h1>Update log</h1>
        <div class="qa">
            <h3 class="question">Added features</h3>
            <div class="answer">
			<p>Sat Feb 29 21:47:50 2020</p><p> update tutorial and FAQ                               </p>
<p>Sat Feb 29 21:47:37 2020</p><p> Add hint info for empty regulons                      </p>
<p>Sat Feb 29 21:47:07 2020</p><p> New submit page layout                                </p>
<p>Sat Feb 29 20:49:00 2020</p><p> Fix: changed CT-R name finding TADs                   </p>
<p>Sat Feb 29 15:49:34 2020</p><p> update sumit page layout                              </p>
<p>Sat Feb 29 15:49:20 2020</p><p> add high charts                                       </p>
<p>Sat Feb 29 15:49:13 2020</p><p> update home layout                                    </p>
<p>Sat Feb 29 11:39:18 2020</p><p> Fix predict label umap                                </p>
<p>Sat Feb 29 00:19:33 2020</p><p> fix parameter cannot be string format                 </p>
<p>Sat Feb 29 00:19:17 2020</p><p> fix marker gene sorting error                         </p>
<p>Sat Feb 29 00:19:01 2020</p><p> update menu color                                     </p>
<p>Sat Feb 29 00:18:54 2020</p><p> fix CT-R name                                         </p>
<p>Sat Feb 29 00:18:41 2020</p><p> update module file                                    </p>
<p>Fri Feb 28 17:33:16 2020</p><p> update CT-R name                                      </p>
<p>Fri Feb 28 17:33:00 2020</p><p> fix color mapping                                     </p>
<p>Fri Feb 28 16:40:05 2020</p><p> update email year                                     </p>
<p>Fri Feb 28 16:33:32 2020</p><p> rename CTS-R                                          </p>
<p>Fri Feb 28 16:32:53 2020</p><p> Merge branch 'update_plot_method'                     </p>
<p>Fri Feb 28 16:32:11 2020</p><p> update                                                </p>
<p>Fri Feb 28 15:24:01 2020</p><p> add highcharts.js                                     </p>
<p>Fri Feb 28 00:21:56 2020</p><p> -pdate: color key on trajectory                       </p>
<p>Thu Feb 27 23:59:45 2020</p><p> -pdate:plot regulon and trajectory method niubility.  </p>
<p>Thu Feb 27 22:53:55 2020</p><p> move general information                              </p>
<p>Thu Feb 27 21:21:20 2020</p><p> iteration set to 1000 if file size too large          </p>
<p>Thu Feb 27 21:21:01 2020</p><p> update submit page layout                             </p>
<p>Thu Feb 27 21:20:52 2020</p><p> update submit page image                              </p>
<p>Thu Feb 27 15:37:59 2020</p><p> prettier format                                       </p>
<p>Thu Feb 27 15:17:58 2020</p><p> sync server submit page version                       </p>
<p>Thu Feb 27 15:17:39 2020</p><p> fix server name                                       </p>
<p>Thu Feb 27 15:17:24 2020</p><p> update home page                                      </p>
<p>Thu Feb 27 15:17:13 2020</p><p> add default umap plot when errors                     </p>
<p>Thu Feb 27 15:15:30 2020</p><p> update home page layout and colors                    </p>
<p>Wed Feb 19 16:14:42 2020</p><p> add default trajectory plot image                     </p>
<p>Wed Feb 19 16:14:29 2020</p><p> fix loading qubic2 params                             </p>
<p>Wed Feb 19 16:04:38 2020</p><p> Remove unused variables                               </p>
<p>Wed Jan 29 16:46:51 2020</p><p> Merge remote-tracking branch 'origin/master'          </p>
<p>Wed Jan 29 16:46:48 2020</p><p> Update: display trajectory plot on seurat cluster     </p>
<p>Wed Dec 18 22:37:53 2019</p><p> Fix: empty custom module                              </p>
<p>Wed Dec 18 13:38:45 2019</p><p> Fix: empty variable                                   </p>
<p>Wed Dec 18 13:38:12 2019</p><p> Fix: number of cells result                           </p>
<p>Wed Dec 18 13:37:42 2019</p><p> Fix: download pdf plot                                </p>
<p>Wed Dec 18 13:36:55 2019</p><p> Fix: sankey plot color                                </p>
<p>Mon Dec 16 16:17:26 2019</p><p> Add: loading result page                              </p>
<p>Mon Dec 16 09:30:27 2019</p><p> Fix: optimized preview expression table               </p>
<p>Mon Dec 16 09:30:02 2019</p><p> Fix: display species on loading result page           </p>
<p>Mon Dec 16 09:29:43 2019</p><p> Fix: include global variable                          </p>
<p>Sat Dec 14 01:30:12 2019</p><p> Fix: normalization and cluster on very small dataset  </p>
<p>Sat Dec 14 01:29:40 2019</p><p> Fix: typo                                             </p>
<p>Wed Dec 11 17:07:49 2019</p><p> Add: generate tomtom result file in pipeline          </p>
<p>Tue Dec 10 16:42:18 2019</p><p> Update README.md                                      </p>
<p>Tue Dec 10 16:41:47 2019</p><p> Fix: working directory path                           </p>
<p>Tue Dec 10 16:41:19 2019</p><p> Update: email notification content                    </p>
<p>Tue Dec 10 16:41:03 2019</p><p> Update: README links                                  </p>
<p>Tue Dec 10 16:40:51 2019</p><p> Remove unused asset                                   </p>
<p>Fri Dec 6 15:51:10 2019 </p><p> init                                                  </p>
<p>Fri Dec 6 15:14:48 2019 </p><p> Initial commit                                        </p>

                <p>Thu Nov 7 15:22:12 2019 </p>
                <p>Add: Recommend fast version</p>
                <p>Thu Nov 7 11:01:17 2019 </p>
                <p>Add: example files</p>
                <p>Tue Nov 5 05:58:13 2019 </p>
                <p>Add: re-run for updating figures</p>
                <p>Tue Nov 5 05:57:11 2019 </p>
                <p>Add: reverse color on predicted label if provide user's label</p>
                <p>Mon Nov 4 15:45:34 2019 </p>
                <p>Add: Test enrichment for paper figures</p>
                <p>Mon Nov 4 15:45:23 2019 </p>
                <p>Add: Count dorothea overlaps for paper figures</p>
                <p>Mon Nov 4 15:45:06 2019 </p>
                <p>Add: RAS heatmap for paper figures</p>
                <p>Sun Nov 3 14:28:27 2019 </p>
                <p>Add: pipepline generate RSS rank scatter plot</p>
                <p>Tue Oct 29 13:29:19 2019 </p>
                <p>Add: install ggplot2</p>
                <p>Tue Oct 29 13:28:31 2019 </p>
                <p>Add: generate rss scatter plot</p>
                <p>Tue Oct 29 13:24:56 2019 </p>
                <p>Add: generate rss scatter plot</p>
                <p>Sun Oct 27 22:59:58 2019 </p>
                <p>Add: display marker genes table</p>
                <p>Sun Oct 27 18:16:50 2019 </p>
                <p>Add: Database file from DoRothEA v2</p>
                <p>Fri Oct 25 17:03:50 2019 </p>
                <p>Add: include list.js library</p>
                <p>Fri Oct 25 15:21:26 2019 </p>
                <p>Add: remove regulon datatabse for memory issue</p>
                <p>Thu Oct 24 11:41:33 2019 </p>
                <p>Add: limit regulon motif list display</p>
                <p>Thu Oct 24 10:11:52 2019 </p>
                <p>Add: example to calculate RSS from your gene list</p>
                <p>Sun Oct 20 15:34:47 2019 </p>
                <p>Add: option for enable imputation</p>
                <p>Sun Oct 20 14:29:08 2019 </p>
                <p>Add: save UMAP colored by predict cell type</p>
                <p>Sun Oct 20 14:09:03 2019 </p>
                <p>Add: remove unused motifs</p>
                <p>Sun Oct 20 12:17:02 2019 </p>
                <p>Add: Button to calculate DoRothEA database overlap</p>
                <p>Sun Oct 20 12:08:51 2019 </p>
                <p>Add: get dorothea database overlap</p>
                <p>Sun Oct 20 12:07:56 2019 </p>
                <p>Add: calculate dorothea database overlap</p>
                <p>Fri Oct 18 10:44:00 2019 </p>
                <p>Add: re-design regulon-TF section</p>
                <p>Thu Oct 3 19:46:31 2019 </p>
                <p>Add: display p-value in scientific notation</p>
                <p>Thu Oct 3 19:16:12 2019 </p>
                <p>Add: Click button to smooth scroll down to image section</p>
                <p>Tue Sep 24 14:37:19 2019 </p>
                <p>Add: support uploading mtx format file</p>
                <p>Tue Sep 24 14:37:01 2019 </p>
                <p>Add: Alternative regulon heatmap link</p>
                <p>Tue Sep 24 14:20:13 2019 </p>
                <p>Add: filter bicluster after imputation</p>
                <p>Tue Sep 24 14:09:57 2019 </p>
                <p>Add: put high value points on top</p>
                <p>Fri Sep 20 13:46:08 2019 </p>
                <p>Add: test weight RSS methods</p>
                <p>Thu Sep 19 08:51:34 2019 </p>
                <p>add: heatmap test</p>
                <p>Thu Sep 19 08:41:04 2019 </p>
                <p>add:test pipeline for 500 cell randomly choosen from hazem 2</p>
                <p>Fri Sep 13 23:12:40 2019 </p>
                <p>Add: test pheatmap for a gene list</p>
                <p>Fri Sep 13 14:17:16 2019 </p>
                <p>add: Pipeline_Test.R : plot module heatmap</p>
                <p>Fri Sep 13 07:46:09 2019 </p>
                <p>add: Yuzhou_test.R Pipeline_Test.R no change with main file</p>
                <p>Fri Sep 13 03:17:38 2019 </p>
                <p>Add: LTMG replace my.object@data</p>
                <p>Fri Sep 13 00:51:59 2019 </p>
                <p>Add: get alternative regulon table and heatmap</p>
                <p>Thu Sep 12 23:44:46 2019 </p>
                <p>undate: add species information to remove mt-gene</p>
                <p>Thu Sep 12 08:01:16 2019 </p>
                <p>Add: q-value thres=0.05</p>
                <p>Wed Sep 11 17:36:18 2019 </p>
                <p>Test: add pheatmap for selected genes</p>
                <p>Wed Sep 11 09:00:30 2019 </p>
                <p>Test: add pheatmap and atac overlap</p>
                <p>Tue Sep 10 01:17:38 2019 </p>
                <p>Add: test generate alternative regulons</p>
                <p>Sat Sep 7 22:51:58 2019 </p>
                <p>Add: fit point size function adjust from test datasets</p>
                <p>Sat Sep 7 17:50:22 2019 </p>
                <p>Add: Rename 10X gene-barcode matrix Update: Plot UMAP cluster (Point size, # of cells)</p>
                <p>Sat Sep 7 17:49:17 2019 </p>
                <p>Add: rename 10X gene-barcodes file</p>
                <p>Fri Sep 6 10:38:34 2019 </p>
                <p>Add: RSS pvalue</p>
                <p>Fri Sep 6 07:49:12 2019 </p>
                <p>Add: merge regulon by TOMTOM matched TF result</p>
                <p>Fri Sep 6 00:38:13 2019 </p>
                <p>add: Multiple_data.R(function complete, usable) change: IRIS3.v3.R work directory change: Yuzhou_Test.R work directory</p>
                <p>Thu Sep 5 17:06:30 2019 </p>
                <p>Add: parallel RSS calculation</p>
                <p>Thu Sep 5 01:58:34 2019 </p>
                <p>Add: test bootstrap p-value</p>
                <p>Sun Sep 1 21:22:28 2019 </p>
                <p>Add: line break on �regulon score�</p>
                <p>Sun Sep 1 21:06:05 2019 </p>
                <p>Add: new plot for RSS scatter</p>
                <p>Sun Sep 1 12:34:06 2019 </p>
                <p>update get regulon add multiple data integration</p>
                <p>Sun Sep 1 02:57:25 2019 </p>
                <p>Add: new RAS method using Wilcox rank sum test</p>
                <p>Sun Sep 1 02:56:38 2019 </p>
                <p>Add: scatter plot RSS for a CT</p>
                <p>Fri Aug 30 17:24:14 2019 </p>
                <p>Add: Cario package to store image</p>
                <p>Fri Aug 30 17:23:42 2019 </p>
                <p>Add: new RAS method from Ranked U-test</p>
                <p>Tue Aug 27 16:49:37 2019 </p>
                <p>Add: plot PCC of Pvalue, Zscore minus RSS</p>
                <p>Mon Aug 26 00:36:41 2019 </p>
                <p>Add: Combine TOMTOM results to TF-Motif list</p>
                <p>Sat Aug 24 22:47:05 2019 </p>
                <p>Add: test regulon rss distribution if have marker genes</p>
                <p>Tue Aug 20 09:44:10 2019 </p>
                <p>Add: new trajectory</p>
                <p>Tue Aug 20 09:43:56 2019 </p>
                <p>Add: new trajectory function</p>
                <p>Mon Aug 19 23:57:56 2019 </p>
                <p>Add: sticky CT selection nav bar</p>
                <p>Sun Aug 18 23:27:41 2019 </p>
                <p>Add: test RSS for repressed regulons</p>
                <p>Sun Aug 18 22:04:37 2019 </p>
                <p>add new plot feature. add regulon cell trajectory</p>
                <p>Sun Aug 18 04:53:43 2019 </p>
                <p>Add: test motif overlap plot with dabestr</p>
                <p>Sat Aug 17 20:53:36 2019 </p>
                <p>Add: Data pre-processing if using 10X data</p>
                <p>Sat Aug 17 11:40:26 2019 </p>
                <p>add lineage line to regulon trajectory</p>
                <p>Sat Aug 17 00:36:19 2019 </p>
                <p>add: release memory for unused variable</p>
                <p>Sat Aug 17 00:35:41 2019 </p>
                <p>add: only perform normalization if all values are integer count</p>
                <p>Sat Aug 17 00:34:47 2019 </p>
                <p>add: process gene symbol with format �gene|chr|##�</p>
                <p>Wed Aug 14 17:40:44 2019 </p>
                <p>add new them to plot</p>
                <p>Wed Aug 14 17:39:37 2019 </p>
                <p>add: coord_fixed</p>
                <p>Wed Aug 14 17:39:21 2019 </p>
                <p>add: imputation, normalization, filtering</p>
                <p>Wed Aug 14 17:21:36 2019 </p>
                <p>add <code>add.line</code> in Plot.Cluster.Trajectry.</p>
                <p>Wed Aug 14 17:19:51 2019 </p>
                <p>add parameter �add.line� in Plot.Cluster.Trajectory.</p>
                <p>Wed Aug 14 11:48:43 2019 </p>
                <p>add: imputation from DrImpute, normalization from scater</p>
                <p>Wed Aug 14 08:18:16 2019 </p>
                <p>add: script to process raw GSE data</p>
                <p>Wed Aug 14 04:49:53 2019 </p>
                <p>add: coord_fixed(1) for all output image</p>
                <p>Wed Aug 14 04:39:47 2019 </p>
                <p>add: restore seurat v3 function</p>
                <p>Wed Aug 14 04:39:28 2019 </p>
                <p>add: hide regulon table head</p>
                <p>Tue Aug 13 17:42:04 2019 </p>
                <p>add option for hdf5</p>
                <p>Thu Aug 8 07:51:24 2019 </p>
                <p>add regulon t-SNE plot</p>
                <p>Wed Aug 7 14:43:32 2019 </p>
                <p>add plot gene t-sne button</p>
                <p>Wed Aug 7 13:03:54 2019 </p>
                <p>add trajectory to pipeline</p>
                <p>Wed Aug 7 11:43:24 2019 </p>
                <p>add gene tsne plot</p>
                <p>Fri Aug 2 21:43:05 2019 </p>
                <p>add plot trajectory</p>
                <p>Wed Jul 31 17:14:39 2019 </p>
                <p>add preview for HDF format</p>
                <p>Tue Jul 30 17:33:16 2019 </p>
                <p>add file format check</p>
                <p>Mon Jul 29 16:53:47 2019 </p>
                <p>add test motif overlap frequency script</p>
                <p>Sun Jul 28 23:41:43 2019 </p>
                <p>add put marker genes on top of regulons</p>
                <p>Sun Jul 28 18:46:26 2019 </p>
                <p>add monocle3.R as official script, add monocle3_test.R for alpha test version</p>
                <p>Fri Jul 26 14:47:40 2019 </p>
                <p>add test result page</p>
                <p>Wed Jul 24 16:07:53 2019 </p>
                <p>add pvalue threshold for removing ras</p>
                <p>Sun Jun 30 21:55:24 2019 </p>
                <p>add sort function for marker gene</p>
                <p>Sun Jun 30 21:54:48 2019 </p>
                <p>add sort marker gene column</p>
                <p>Sun Jun 30 11:54:14 2019 </p>
                <p>add parameter options from sc3</p>
                <p>Fri Jun 28 07:58:24 2019 </p>
                <p>add more methods calculating regulon activity score and sorting</p>
                <p>Thu Jun 27 17:22:04 2019 </p>
                <p>add custom methods plot</p>
                <p>Thu Jun 27 17:21:25 2019 </p>
                <p>add multiple score methods</p>
                <p>Thu Jun 27 09:14:11 2019 </p>
                <p>Merge pull request #1 from Wang-Cankun/add-license-1</p>
                <p>Mon Jun 24 22:26:51 2019 </p>
                <p>add sort p-val and zscore</p>
                <p>Mon Jun 24 16:33:28 2019 </p>
                <p>add AUC score</p>
                <p>Thu Jun 20 22:56:17 2019 </p>
                <p>add plot functions</p>
                <p>Thu Jun 20 16:54:09 2019 </p>
                <p>IRIS3.v2.R Add unique gene finding function</p>
                <p>Thu Jun 20 16:46:33 2019 </p>
                <p>IRIS3.v2.R add gene marker finging function</p>
                <p>Thu Jun 20 16:44:29 2019 </p>
                <p>IRIS3.v2.R add marker gene select function</p>
                <p>Thu Jun 20 15:49:46 2019 </p>
                <p>add auto disable heatmap</p>
                <p>Wed Jun 19 17:46:51 2019 </p>
                <p>add version 2</p>
                <p>Wed Jun 19 15:54:43 2019 </p>
                <p>add test AUC code</p>
                <p>Mon Jun 17 15:15:27 2019 </p>
                <p>add 2 tsne plot</p>
                <p>Mon Jun 17 15:15:20 2019 </p>
                <p>add 3 example types</p>
                <p>Mon Jun 17 15:15:01 2019 </p>
                <p>add column for gene id</p>
                <p>Mon Jun 17 10:02:42 2019 </p>
                <p>add convert meme</p>
                <p>Thu Jun 13 17:10:13 2019 </p>
                <p>add required packages</p>
                <p>Wed Jun 12 16:59:34 2019 </p>
                <p>add gene id next to draw scale</p>
                <p>Tue Jun 11 11:39:23 2019 </p>
                <p>add button on regulon tsne</p>
                <p>Tue Jun 11 00:21:59 2019 </p>
                <p>add button display regulon tsne</p>
                <p>Mon Jun 10 16:31:31 2019 </p>
                <p>remove cellity, add test dir</p>
                <p>Mon Jun 10 16:30:50 2019 </p>
                <p>Revert �remove cellity, add test dir�</p>
                <p>Mon Jun 10 16:30:20 2019 </p>
                <p>remove cellity, add test dir</p>
                <p>Mon Jun 3 19:12:01 2019 </p>
                <p>add background genome</p>
                <p>Mon Jun 3 16:43:19 2019 </p>
                <p>add button to save svg,png format</p>
                <p>Fri May 24 10:24:18 2019 </p>
                <p>change Dr. Qin Ma�s email address</p>
                <p>Fri May 24 10:24:00 2019 </p>
                <p>add max cores</p>
                <p>Tue Apr 16 05:57:23 2019 </p>
                <p>add warning for count 0 values in 1000 lines.</p>
                <p>Mon Apr 15 06:16:46 2019 </p>
                <p>increase preview table padding</p>
                <p>Sat Apr 13 07:29:08 2019 </p>
                <p>add error msg for not enough cell number</p>
                <p>Mon Apr 8 05:24:50 2019 </p>
                <p>add sort p-value for TOMTOM motifs</p>
                <p>Mon Apr 8 00:05:51 2019 </p>
                <p>add datatables.button, lozad.js</p>
                <p>Sat Apr 6 16:40:27 2019 </p>
                <p>add txt file to repo</p>
                <p>Sat Apr 6 16:22:12 2019 </p>
                <p>add selection for multi species</p>
                <p>Thu Apr 4 04:00:17 2019 </p>
                <p>add gene filter option</p>
                <p>Thu Apr 4 03:56:53 2019 </p>
                <p>add atac-seq enrichment table</p>
                <p>Fri Mar 29 03:41:52 2019 </p>
                <p>add option for user determine k in sc3</p>
                <p>Fri Mar 29 03:41:31 2019 </p>
                <p>add gene modult tab</p>
                <p>Wed Mar 27 22:07:04 2019 </p>
                <p>add gene module process</p>
                <p>Mon Mar 25 17:20:37 2019 </p>
                <p>add error message when bicluster not enough</p>
                <p>Wed Feb 13 03:32:13 2019 </p>
                <p>add sitemap</p>
                <p>Wed Feb 13 03:30:55 2019 </p>
                <p>add script sort bbc fasta file</p>
                <p>Wed Feb 13 03:30:39 2019 </p>
                <p>add output sorted pvalue fasta header list</p>
                <p>Fri Feb 1 03:14:48 2019 </p>
                <p>add a tab of more page</p>
                <p>Fri Feb 1 03:14:11 2019 </p>
                <p>add a error page</p>
                <p>Fri Feb 1 03:13:56 2019 </p>
                <p>add a tab of more</p>
                <p>Fri Feb 1 03:13:31 2019 </p>
                <p>add faq page for test</p>
                <p>Fri Feb 1 03:13:04 2019 </p>
                <p>add a more page for faq, citation and contact</p>
                <p>Sat Dec 29 17:49:29 2018 </p>
                <p>add background script vanta.js</p>
                <p>Sat Dec 29 09:01:05 2018 </p>
                <p>add SC3 consensus matrix</p>
                <p>Sat Dec 29 09:00:45 2018 </p>
                <p>add user�s label if have</p>
                <p>Sat Dec 29 03:21:07 2018 </p>
                <p>add parameter info</p>
                <p>Fri Dec 28 08:33:26 2018 </p>
                <p>add hint for uploading gene symbols, remove c_arg=0.5</p>
                <p>Fri Dec 28 08:33:00 2018 </p>
                <p>add function send to enrichr</p>
                <p>Fri Dec 28 05:17:25 2018 </p>
                <p>add style assets</p>
                <p>Fri Dec 28 05:08:00 2018 </p>
                <p>add tools</p>
                <p>Fri Dec 28 05:06:51 2018 </p>
                <p>add png</p>
                <p>Fri Dec 28 03:14:49 2018 </p>
                <p>add test case on meme motif</p>
                <p>Fri Dec 28 03:14:10 2018 </p>
                <p>add option to test meme motif</p>
                <p>Fri Dec 28 03:12:50 2018 </p>
                <p>add covert_meme function</p>
                <p>Fri Dec 28 03:11:44 2018 </p>
                <p>add option for meme and gene filtering</p>
                <p>Thu Dec 27 07:02:36 2018 </p>
                <p>fix add silh score</p>
                <p>Thu Dec 27 02:02:06 2018 </p>
                <p>add option for using filter</p>
                <p>Wed Dec 26 20:33:12 2018 </p>
                <p>add clustergrammer</p>
                <p>Tue Dec 25 21:54:47 2018 </p>
                <p>add a regulon_motif file specify which regulon belongs to</p>
                <p>Tue Dec 25 21:54:01 2018 </p>
                <p>add script generating motif logo and tomtom</p>
                <p>Tue Dec 25 21:52:43 2018 </p>
                <p>add ari score if both have sc3 and label option</p>
                <p>Mon Dec 24 15:53:04 2018 </p>
                <p>add info</p>
                <p>Mon Dec 24 15:52:57 2018 </p>
                <p>add info</p>
                <p>Sat Dec 22 21:20:58 2018 </p>
                <p>add score raw data</p>
                <p>Wed Dec 19 15:19:11 2018 </p>
                <p>add sc3 pipeline</p>
                <p>Sun Dec 16 14:40:14 2018 </p>
                <p>add category for each ct</p>
                <p>Sun Dec 16 06:57:11 2018 </p>
                <p>add function check species</p>
                <p>Thu Dec 13 14:54:56 2018 </p>
                <p>add feature remove false positive regulon</p>
                <p>Thu Dec 13 07:54:08 2018 </p>
                <p>add test data for motif visualization</p>
                <p>Sun Dec 9 05:08:45 2018 </p>
                <p>add motif tools</p>
                <p>Sun Dec 2 05:19:57 2018 </p>
                <p>add regulon category to column</p>
                <p>Mon Nov 26 09:31:52 2018 </p>
                <p>add prepare heatmap</p>
                <p>Sun Nov 25 07:11:27 2018 </p>
                <p>Add email notification</p>

            </div>
            <div class="clear"></div>
        </div>
        <!--
		<h3 class="question">Mar 2019</h3>
		<div class="answer">
			<h4>03/31/2019</h4>
			<p><b>[Added]</b> Additional function for ATAC-Seq peak enrichment.</p>
			<p><b>[Added]</b> Additional function to retrive genes outside CTS-Rs through topologically associated domain (TAD) comparision </p>
			<h4>03/25/2019</h4>
			<p><b>[Added]</b> Users can now upload their own gene modules to IRIS3 and perform analysis along with the expression matrix.</p>
			<p><b>[Changed]</b> Use Monocle to replace TPM normalization method. </p>
			<p><b>[Changed]</b> Change the threshold of gene filtering from 96% to 95%. Added cell filtering function to remove genes with zero values more than 99%. Both gene and cell filtrations are set to be ON as default.</p>
		</div>
		<div class="clear"></div>
	</div>
		
		<h3 class="question">Dec 2018</h3>
		<div class="answer">
			<h4>12/23/2018</h4>
			<p><b>[Added]</b> Homepage background effect using <a href="https://www.vantajs.com/" target="_blank"> Vanta.js</a> </p>
			<p><b>[Changed]</b> Page Layout, link and button color improvements. </p>
			<p><b>[Changed]</b> Website load time has been optimized 40% based on the report from <a href="https://gtmetrix.com/" target="_blank"> GTmetrix</a> 
			and <a href="https://developers.google.com/web/tools/lighthouse/" target="_blank"> Lighthouse.</a></p>
		</div>
		<div class="clear"></div>
	</div>
	
	<div class="qa">
		<h3 class="question">Nov 2018</h3>
		<div class="answer">
			 <h4>11/06/2018</h4>
			<p><b>[Added]</b><a href="https://clustergrammer.readthedocs.io/" target="_blank"> Clustergrammer</a>, 
			a interactive web-based tool for visualizing heatmaps for result visualization.	</p>
			<p><b>[Added]</b><a href="https://bmbl.bmi.osumc.edu/IRIS3/assets/img/IRIS3_pipeline.png" target="_blank"> Pipeline overview</a> on homepage.</p>
			<p><b>[Added]</b><a href="https://bmbl.bmi.osumc.edu/IRIS3/tutorial.php#1basics" target="_blank"> Tutorial</a>,<a href="https://bmbl.bmi.osumc.edu/IRIS3/news.php#1version" target="_blank"> What's new</a>,
			<a href="https://groups.google.com/forum/#!forum/IRIS3-discussion-board" target="_blank">Forum</a> sections.</p>
			
			<h4>11/06/2018</h4>
			<p> The website is now online.</p>
		</div>
		
		<div class="clear"></div>
	</div>	
	-->
    </div>
</body>

</html>