<html class="gr__amp_pharm_mssm_edu">

<head>
    <meta http-equiv="Content-Type" content="text/html; charset=windows-1252">
    <style type="text/css" abt="234"></style>
</head>

<body data-gr-c-s-loaded="true">
    <div id="the-basics-content">
        <h1>Example result illustration</h1>

        <div class="answer">
            <br>
            <p>We applied IRIS3 on a single-cell RNA-seq dataset consisting of 3005 cells from Mouse lung epithelial from
                <a href="https://science.sciencemag.org/content/347/6226/1138"><b>Zeisel, et al.</b></a>. Default parameters (k=20, o=500, f=0.7) were used for QUBIC biclustering; Dual strategy was enabled, imputation was disabled. Cell clusters were predicted
                using Seurat and evaluated by comparing with the ground-truth labels provided by the paper; regulons were inferenced using DMINDA and MEME based on the clustering result.</p>

            <p>This tutorial will guide you through the output page of IRIS3 in details.</p>

        </div>
        <div class="clear"></div>

        <div class="qa">
            <h3 class="question">Clusters and general information</h3>
            <div class="answer">
                <p><img src="./assets/img/tutorial/3.1.1.jpg" alt="" style="width:100%"></p>
				
                <ul>
                    
                    <li>The clustering result includes two parts:  an interactive UMAP and a table. A floating window will show up when move your mouse on cells, indicating the belonging cell cluster and cell name. The table shows the overall cell number and regulon number in each cluster.</li>
                </ul>
				<div class="clear"></div>
				<p><img src="./assets/img/tutorial/3.1.2.jpg" alt="" style="width:100%"></p>
				<div class="row"></div>
				<br/>
				<b>A. clustering evaluation: </b>
				<ul>
                    <li> A table showing the four Rand indexes for the predicted cell type evaluations.
					</li>
                    <li>Four index scores are used to evaluate the closeness of the predicted</li>
                    <li>Two UMAP plots show the benchmark cell label and predicted cell clusters, respectively.</li>
                    <li>A trajectory plot is generated from Slingshot integrated in IRIS3.</li>
					<li>A bar plot showing the silhouette score that measures how similar each cell is to its type compared to other clusters.</li>
					<li>A Sankey plot showing the convergence and divergence of a two cell groups (ground truth versus Seurat prediction). The width of the arrows shown proportionally to the flow quantity.</li>
					<li>Note: if ground-truth cell labels are not provided by the user, the Score table and Sankey plot will be omitted.</li>
                </ul>
				<b>B. marker genes:</b>
				<ul>
                    <li> The top 100 marker genes in each cluster generated from the differential expression analysis using Seurat.
					</li>
                </ul>
				<b>C. job information:</b>
				<ul>
                    <li> Listing pre-processing information and job settings.
					</li>
                </ul>
            </div>
            <div class="clear"></div>
        </div>
        <div class="qa">
            <h3 class="question">Regulon heatmap and RSS distribution</h3>
            <div class="answer">
                <img src="./assets/img/tutorial/3.1.3.jpg" style="width:100%">
				<br/>
				<div class="row"></div>
                <p>Following the general results, a cell-gene-regulon heatmap is displayed to cover the gene distribution of regulons in each cell type and corresponding expression level across all cells. </p>
                <p>To explore details about Clustergrammer, please read the <a href="https://clustergrammer.readthedocs.io/"><b>documentation</b></a>.
                    <br>
                    <ul>
                        <li>(A) The heatmap, empowered by Clustergrammer, showcased the expression pattern of genes from the top ten regulons in the corresponding cell type. Users can rearrange the columns and rows by grouping genes in one regulon or cells in a
                            cell type. Genes included in a regulon are labeled as green. The genes can be sorted by each regulon by double-clicking the regulon name. Due to the space limit, at most 10 regulons are shown in the figure based on the <a href="https://bmbl.bmi.osumc.edu/iris3/tutorial.php#4FAQ&q=7"><b>rank of regulons</b></a></li>
                        <li>(B) The scatter plot shows the distribution of RSS in cell type 1. The top five regulons are highlighted with TF names.</li>
                    </ul>

            </div>
            <div class="clear"></div>
        </div>

        <div class="qa">
            <h3 class="question">Regulon results in detail</h3>
            <div class="answer">
				<img src="./assets/img/tutorial/3.1.4.jpg" style="width:100%">
				<br/>
				<div class="row"></div>
                <ol>
				<li>Each regulon is named by the cell cluster index and regulon number. A regulon is a cell-type-specific regulon (CTSR) if the p-value of regulon specificity score (RSS) is less than 0.05.  </li>
				<li>Stars indicate marker genes, that is, the differential expressed gene, identified in each cluster using Seurat. </li>
				<li>Gene symbols and link to the GeneCards (Human) or the Mouse Genome Informatics (MGI) website. </li>
				<li>Corresponding gene Ensembl ID and link to the website. </li>
				<li>Gene expression UMAP and comparison to the cell types. </li>
				<li>The corresponding TF with a corresponding link to HOCOMOCO database.  </li>
				<li>Detailed motif finding results from including positions, sequences, position weight matrix, etc. </li>
				<li>Motif details linking to the TOMTOM database. </li>
				<li>A Clustergrammer heatmap showing the expression value of all genes Of this regulon and cell type. </li>
				<li>EnrichR link to the enrichment analysis of this regulon.  </li>
				<li>Bulk ATAC peak enrichment test. <a href="https://bmbl.bmi.osumc.edu/iris3/more.php#4FAQ&q=7">See details here.</a> </li>
				<li>Coverage of regulon genes and the topologically associated regions. <a href="https://bmbl.bmi.osumc.edu/iris3/more.php#4FAQ&q=8">See details here.</a> </li>
				<li>The regulon activity UMAP and comparison to the cell types. </li>
				<li>Trajectory analysis colored by cell type and regulon activity, respectively. </li>
				</ol>


            </div>
            <div class="clear"></div>
        </div>



    </div>
</body>

</html>