<html class="gr__amp_pharm_mssm_edu">

<head>
    <meta http-equiv="Content-Type" content="text/html; charset=windows-1252">
    <style type="text/css" abt="234"></style>
</head>

<body data-gr-c-s-loaded="true">
    <div id="the-basics-content">
        <h1>Example result illustration</h1>

        <div class="answer">
            <br>
            <p>We applied IRIS3 on a single-cell RNA-seq dataset consisting of 3005 cells from Mouse lung epithelial from
                <a href="https://science.sciencemag.org/content/347/6226/1138"><b>Zeisel, et al.</b></a>. Default parameters (k=20, o=500, f=0.7) were used for QUBIC biclustering; Dual strategy was enabled, imputation was disabled. cell types were predicted
                using Seurat and evaluated by comparing with the ground-truth labels provided by the paper; CTSRs were inferenced using DMINDA and MEME based on the clustering result.</p>

            <p>This tutorial will guide you through the output page of IRIS3 in details.</p>

        </div>
        <div class="clear"></div>

        <div class="qa">
            <h3 class="question">General results</h3>
            <div class="answer">
                <p><img src="./assets/img/tutorial/3.1.1.png" alt=""></p>
                <ul>
                    <li>Click to download a zip file containing the result files. See <a href="https://bmbl.bmi.osumc.edu/iris3/more.php#4FAQ&amp;q=12"><strong>here</strong></a> for details.</li>
                    <li>including: species, the number of original cells and genes, the number of removed (filtered) cells and genes as well as their filtering ratio, the number of provided cell type by users, the number of predicted cell types, the total
                        number of predicted biclusters (gene modules), and the total number of identified CTSRs.</li>
                </ul>

            </div>
            <div class="clear"></div>
        </div>
        <div class="qa">
            <h3 class="question">Cell type prediction results</h3>
            <div class="answer">
                <p><img src="./assets/img/tutorial/3.1.2.png" alt=""></p>
                <ul>
                    <li>A table showing the <a href="https://bmbl.bmi.osumc.edu/iris3/more.php#4FAQ&amp;q=3"><strong>four Rand indexes</strong></a> for the predicted cell type evaluations.</li>
                    <li>Four index scores are used to evaluate the closeness of the predicted</li>
                    <li>cell type to the benchmark. Two UMAPs show the benchmark cell label and predicted cell clusters, respectively.</li>
                    <li>A trajectory plot is generated from Slingshot integrated in IRIS3.</li>
                    <li>A bar plot showing the silhouette score that measures how similar each cell is to its type compared to other clusters.</li>
                    <li>A Sankey plot showing the convergence and divergence of a two cell groups (ground truth versus Seurat prediction). The width of the arrows shown proportionally to the flow quantity.</li>
                </ul>
                <p><strong>Note:</strong> if ground-truth cell labels are not provided by the user, the Score table and Sankey plot will be omitted.</p>


            </div>
            <div class="clear"></div>
        </div>

        <div class="qa">
            <h3 class="question">Marker genes results</h3>
            <div class="answer">
                <img src="./assets/img/tutorial/3.1.3.png">
                <p>A complete marker gene list generated from differential expression analysis using Seurat.</p>
                <br>
            </div>
            <div class="clear"></div>
        </div>

        <div class="qa">
            <h3 class="question">Job settings</h3>
            <div class="answer">
                <img src="./assets/img/tutorial/3.1.4.png">
                <p>A table showing all settings and file names of this Job.</p>
                <br>
            </div>
            <div class="clear"></div>
        </div>

        <div class="qa">
            <h3 class="question">CTSR Heatmaps in a cell type</h3>
            <div class="answer">
                <img src="./assets/img/tutorial/3.2.1.png" style="width:100%">
                <p>Following the general results, a CTS cell-gene-regulon heatmap is displayed to cover the gene distribution of CTSRs in each cell type and corresponding expression level across all cells. </p>
                <p>To explore details about Clustergrammer, please read the <a href="https://clustergrammer.readthedocs.io/"><b>documentation</b></a>.
                    <br>
                    <ul>
                        <li>(A) The heatmap, empowered by Clustergrammer, showcased the expression pattern of genes from the top ten CTSRs in the corresponding cell type. Users can rearrange the columns and rows by grouping genes in one CTSR or cells in a
                            cell type. Genes included in a CTSR are labeled as green. The genes can be sorted by each CTSR by double-clicking the CTSR name. Due to the space limit, at most 15 CTSRs are shown in the figure based on the <a href="https://bmbl.bmi.osumc.edu/iris3/tutorial.php#4FAQ&q=7"><b>rank of CTSRs</b></a></li>
                        <li>(B) The scatter plot shows the distribution of RSS in cell type 1. The top five CTSRs are highlighted with TF names.</li>
                    </ul>

            </div>
            <div class="clear"></div>
        </div>

        <div class="qa">
            <h3 class="question">Detailed result of CTSRs</h3>
            <div class="answer">
                <img src="./assets/img/tutorial/3.3.1.png" style="width:100%">
                <p>(1) RSS, empirical p-value, and number of genes in the CTSR.<br> (2) Stars indicate marker genes identified from the differential expression analysis.<br> (3) Gene symbols and link to the GeneCards (Human) or the Mouse Genome Informatics
                    (MGI) website.<br> (4) Corresponding gene Ensembl ID and link to the website.<br> (5) Gene expression UMAP and comparison to the cell types.<br> (6) The corresponding TF with a corresponding link to HOCOMOCO database. (7) CTSRs in
                    other cell types that regulated by the same TF.<br> (8) Detailed motif finding information.<br> (9) (10) A Clustergrammer heatmap shows the expression value of all genes in current CTSR and cell type.<br> (11) EnrichR link for gene
                    enrichment analysis of the CTSR.<br> (12) ATAC peak enrichment test.<br> (13) additional analysis to find genes in other CTSRs may in the same topologically associated regions as the current CTSR genes.<br> (14) The regulon activity
                    UMAP and comparison to the cell types.<br> (15) Trajectory analysis colored by cell type and CTSR activity, respectively.</p>


            </div>
            <div class="clear"></div>
        </div>
    </div>
</body>

</html>